freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtRoleComponent.java

```diff
@@ -25,21 +26,25 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(KerberosMgmtRoleComponent.class);
 
-    public boolean privilegesExist(RoleRequest roleRequest, FreeIpaClient ipaClient) {
-        return roleRequest == null ||
-            roleRequest.getPrivileges().stream().allMatch(privilegeName -> {
-                try {
-                    ipaClient.showPrivilege(privilegeName);
-                    return true;
-                } catch (FreeIpaClientException e) {
-                    if (!FreeIpaClientExceptionUtil.isNotFoundException(e)) {
-                        LOGGER.debug("Privilege [{}] does not exist", privilegeName);
-                    } else {
-                        LOGGER.error("Privilege [{}] show error", privilegeName, e);
-                    }
-                    return false;
-                }
-            });
+    public boolean privilegesExist(RoleRequest roleRequest, FreeIpaClient ipaClient) throws FreeIpaClientException {
+        try {
+            return roleRequest == null ||
+                    roleRequest.getPrivileges().stream().allMatch(privilegeName -> {
+                        try {
+                            ipaClient.showPrivilege(privilegeName);
+                            return true;
+                        } catch (FreeIpaClientException e) {
+                            if (!FreeIpaClientExceptionUtil.isNotFoundException(e)) {
+                                LOGGER.error("Privilege [{}] show error", privilegeName, e);
+                                throw new FreeIpaClientExceptionWrapper(e);
```

Comment:
why is it necessary to wrap it?

Reply:
`FreeIpaClientException` is a checked exception, so I wrap it with an unchecked exception which is thrown from the lambda passed into `matchAll()`. I then catch the unchecked exception and rethrow the original exception outside of the lambda.